panic in the github.com/pingcap/tidb/br/pkg/streamhelper.(*FlushSubscriber).Clear

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2a365f3]
goroutine 47 [running]:
github.com/pingcap/tidb/br/pkg/streamhelper.(*FlushSubscriber).Clear(0x0)
	br/pkg/streamhelper/flush_subscriber.go:113 +0x2d3
github.com/pingcap/tidb/br/pkg/streamhelper.(*FlushSubscriber).Drop(...)
	br/pkg/streamhelper/flush_subscriber.go:121
github.com/pingcap/tidb/br/pkg/streamhelper.(*CheckpointAdvancer).stopSubscriber(0xc00047e980)
	br/pkg/streamhelper/advancer.go:518 +0xb8
github.com/pingcap/tidb/br/pkg/streamhelper.(*CheckpointAdvancer).OnStop(0xc00047e980)
	br/pkg/streamhelper/advancer_daemon.go:52 +0x7e
github.com/pingcap/tidb/br/pkg/streamhelper.(*CheckpointAdvancer).OnBecomeOwner.func1()
	br/pkg/streamhelper/advancer_daemon.go:40 +0x54
created by github.com/pingcap/tidb/br/pkg/streamhelper.(*CheckpointAdvancer).OnBecomeOwner in goroutine 163
	br/pkg/streamhelper/advancer_daemon.go:38 +0x113 
```
### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/br/pkg/streamhelper/advancer.go
+++ b/br/pkg/streamhelper/advancer.go
@@ -515,8 +515,10 @@ func (c *CheckpointAdvancer) advanceCheckpointBy(ctx context.Context,
 func (c *CheckpointAdvancer) stopSubscriber() {
 	c.subscriberMu.Lock()
 	defer c.subscriberMu.Unlock()
-	c.subscriber.Drop()
-	c.subscriber = nil
+	if c.subscriber != nil {
+		c.subscriber.Drop()
+		c.subscriber = nil
+	}
 }
 
 func (c *CheckpointAdvancer) SpawnSubscriptionHandler(ctx context.Context) {

